Documentation/network/concepts/masquerading.rst

```diff
@@ -44,6 +44,17 @@ Setting the routable CIDR
 Setting the masquerading interface
   See :ref:`masq_modes` for configuring the masquerading interfaces.
 
+Masquerade traffic to Remote Nodes
+**********************************
+
+To masquerade traffic to remote nodes in BPF masquerading mode, use the option ``enable-remote-node-masquerade: "true"``. 
+This option requires ``enable-bpf-masquerade: "true"`` and also either 
+``enable-ipv4-masquerade: "true"`` or ``enable-ipv6-masquerade: "true"`` to SNAT traffic for IPv4 and IPv6, respectively.
+
+However, this flag does not provide the optimal behavior because packets directed to the nodes' InternalIP addresses will also be masqueraded. 
+To achieve the optimal behavior, we would need to inform the datapath about the type of IP address each node IP address represents. 
+This could potentially be accomplished by extending the node ID map.
+
```

Comment:
I could imagine this feature leading to confused deputy attacks in an environment, so it's probably best recommended to be disabled by default:

```suggestion

This option can limit Cilium's ability to enforce ingress host firewall policies for traffic from Pods towards Nodes.
If you use this option, consider establishing default deny policies to prevent Endpoints from communicating with Nodes,
and minimize use of policy statements that allow traffic from remote nodes (such as the ``remote-node`` Entity).
If in doubt, disable this option.

```